torch/testing/_internal/common_methods_invocations.py

```diff
@@ -86,7 +87,7 @@ def __init__(
             active_if: whether tests matching the above arguments should be skipped
             expected_failure: whether to assert that skipped tests fail
         """
-        decorator = expectedFailure(device_type) if expected_failure else skipIf(True, "Skipped!")
+        decorator = unittest.expectedFailure if expected_failure else unittest.skip("Skipped!")
```

Comment:
Why not wrap the function in the `expectedFailure` decorator?

https://github.com/pytorch/pytorch/blob/565578cdab2cbf18fac0ea97c1a1f954492cc8b5/torch/testing/_internal/common_device_type.py#L890

Like could the `unittest.expectedFailure` decorator be applied to that function before calling it?

Reply:
That won't work. The unittest.expectedFailure is implemented as follows

```python
def expectedFailure(test_item):
    test_item.__unittest_expecting_failure__ = True
    return test_item
```

So wrapping `fn` with it will not change the behavior of the function. We need to expose this attribute to the unittest engine.

Reply:
Aha! You're right. My mistake.